db/migrate/20250606094103_drop_legacy_models.rb

```diff
@@ -0,0 +1,26 @@
+class DropLegacyModels < ActiveRecord::Migration[7.2]
+  def up
+    # Check if data migration is complete
+    if DataMigrationStatus.find_by(name: "push_model_migration")&.completed? ||
+        ActiveRecord::Base.connection.execute(<<-SQL).first["total_count"] == 0
+         SELECT (
+           (SELECT COUNT(*) FROM passwords) +
+           (SELECT COUNT(*) FROM urls) +
+           (SELECT COUNT(*) FROM file_pushes)
+         ) as total_count
+        SQL
```

Comment:
This part will never be true if I don't miss any point. Because `MigrateDataToPushModelJob` is creating new records but not removing old records. So, this migration couldn't be run correctly.

Reply:
For DBs with legacy data where `MigrateDataToPushModelJob` has run successfully, the first condition `DataMigrationStatus.find_by(name: "push_model_migration")&.completed?` will return true and the `drop_table` calls will be run.

For new databases that have never had Legacy data `ActiveRecord::Base.connection.execute` will return true (zero records) and the `drop_table` calls will be run.

The failure case is where a user has legacy data but there is no `DataMigrationStatus` record that is marked `completed=true`.

In that (hopefully) rare case, we raise an exception and force the user to run the previous version until the `MigrateDataToPushModelJob` completes.

Reply:
Got it. I missed `||` part of `if` statement. Good work :rocket: